core/arch/arm/pta/system.c

```diff
@@ -496,6 +496,39 @@ static TEE_Result system_copy_from_ta_binary(struct system_ctx *ctx,
 			    params[0].value.b, params[1].memref.size);
 }
 
+static TEE_Result system_set_prot(struct tee_ta_session *s,
+				  uint32_t param_types,
+				  TEE_Param params[TEE_NUM_PARAMS])
+{
+	const uint32_t accept_flags = PTA_SYSTEM_MAP_FLAG_WRITEABLE |
+				      PTA_SYSTEM_MAP_FLAG_EXECUTABLE;
+	uint32_t exp_pt = TEE_PARAM_TYPES(TEE_PARAM_TYPE_VALUE_INPUT,
+					  TEE_PARAM_TYPE_VALUE_INPUT,
+					  TEE_PARAM_TYPE_NONE,
+					  TEE_PARAM_TYPE_NONE);
+	uint32_t prot = TEE_MATTR_UR | TEE_MATTR_PR;
+	uint32_t flags = 0;
+	vaddr_t va = 0;
+	size_t sz = 0;
+
+	if (exp_pt != param_types)
+		return TEE_ERROR_BAD_PARAMETERS;
+
+	flags = params[0].value.b;
+
+	if ((flags & accept_flags) != flags)
+		return TEE_ERROR_BAD_PARAMETERS;
+	if (flags & PTA_SYSTEM_MAP_FLAG_WRITEABLE)
+		prot |= TEE_MATTR_UW | TEE_MATTR_PW;
```

Comment:
Why `TEE_MATTR_PW`?

Reply:
`TEE_MATTR_PW` is redundant as it's implicit by `TEE_MATTR_UW` in the ARM architecture, but I think it's more clear to be explicit.